update backend guide (#8)
fluencelabs/docs: docs/aqua-book/language/header/header.md

> An Aqua source file has a head and a body. The body contains function definitions, services, types, constants. The header manages what is imported from other files and what is exported.

Reviewer: head => header

> ## Aqua component

Reviewer: What is "component"?
```suggestion
## Aqua source file header
```

> By default, `.aqua` file exports and declares everything it contains. With `aqua` header you can describe the `.aqua` file's interface.

Reviewer: ```suggestion
When header is omitted, `.aqua` file exports and declares everything it contains. With `aqua` header you can control what gets exported from the aqua file.
```

```aqua
-- `aqua` expression may be only on the very first line of the file

> aqua AquaName declares *

Reviewer: ```suggestion
aqua FileName declares *
```

```

`Aqua.Name` may contain dots.

`AquaName` can be used as the aqua's name when this file is `use`d. In this case, only what is enumerated in `declares` section will be available. `declares *` allows you to declare everything in the file as the module interface.

```aqua

> aqua AquaName declares CONST_NAME, ServiceName, MyType, fn

Reviewer: AquaName => FileName
fn => functionName, because you cannot really use `fn` in `declares`, right?
```suggestion
aqua FileName declares CONST_NAME, ServiceName, MyType, functionName
```

## Module

By default, `.aqua` file exports and declares everything it contains. With `aqua` header you can describe the `.aqua` file's interface.

```aqua
-- Module expression may be only on the very first line of the file

> aqua ModuleName declares *

Reviewer 2: Change `Module` here as well
